### PR TITLE
Instructors functions

### PR DESCRIPTION
**Pull request by Omar-Alkhatib**
Functions for Instructor added: 
registeredCoursesInstructor, 

Functions for Admin added: 
    students_details, 
    disable_user, 
    instructors_details

### CODE REVIEW

File: Back-End/controllers/admin_controllers/instructors_controllers.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+const instructors_details = (req, res) => {
+    const query = `SELECT * FROM users WHERE role_id = 2`;
+    db.query(query, (err, result) => {
+        if (err) throw err
+        res.json(result)
+    })
+}
+
 module.exports = {
+    instructors_details
```
Comment (Besslan) [inline Back-End/controllers/admin_controllers/instructors_controllers.js]: Add a trailing comma, also remove the empty line

File: Back-End/controllers/admin_controllers/students_controllers.js [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+    
+    db.query(query, (err, result) => {
+        if (err) throw err
+        res.json(result)
+    })
+}
+
+const disable_user = (req, res) => {
+    const query = `UPDATE users SET isDisabled = 1 WHERE id = ?`
+    const data = req.params.id
```
Comment (Besslan) [inline Back-End/controllers/admin_controllers/students_controllers.js]: No semicolon, make sure to use prettier to format the code,  make sure to format all of the pages you have worked on

File: Back-End/controllers/instructors_controllers/courses_controllers.js [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+    const query = `SELECT * FROM courses c JOIN registration r ON c.id = r.course_id AND instructor_id = ?`;
+    const data = req.params.id;
+    db.query(query, data, (err, result) => {
+        if (err) throw err
+        res.json(result)
+    })
+};
+
+
+
```
Comment (Besslan) [inline Back-End/controllers/instructors_controllers/courses_controllers.js]: These 2 lines are extra, make sure to remove them

File: Back-End/controllers/instructors_controllers/courses_controllers.js [hunk 4]
```diff
@@ -1,5 +1,16 @@
...
 module.exports = {
-   
+    registeredCoursesInstructor
```
Comment (Besslan) [inline Back-End/controllers/instructors_controllers/courses_controllers.js]: no trailing comma, the code formatter can handle this kind of things

File: Back-End/routes/admin.route.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 const adminRouter = require("express").Router();
+const {students_details, disable_user, instructors_details} = require('../controllers/main-controller')
+
```
Comment (Besslan) [inline Back-End/routes/admin.route.js]: no semicolon and an empty line

File: Back-End/controllers/admin_controllers/instructors_controllers.js [hunk 6]
```diff
@@ -1,6 +1,15 @@
 const db = require('../../db');
 
+const instructors_details = (req, res) => {
+    const query = `SELECT * FROM users WHERE role_id = 2`;
```
Comment (Besslan) [inline Back-End/controllers/admin_controllers/instructors_controllers.js]: This query needs to be dynamic, what if changes happen the role_id 2 is used for another role

File: Back-End/controllers/admin_controllers/students_controllers.js [hunk 7]
```diff
@@ -1,6 +1,26 @@
 const db = require('../../db');
 
+
+const students_details = (req, res) => {
+    const query = `SELECT * FROM users WHERE role_id = 3`;
```
Comment (Besslan) [inline Back-End/controllers/admin_controllers/students_controllers.js]: same as the query above

File: Back-End/controllers/admin_controllers/instructors_controllers.js (hunk 1 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/admin_controllers/instructors_controllers.js]: Noted

File: Back-End/controllers/admin_controllers/students_controllers.js (hunk 2 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/admin_controllers/students_controllers.js]: Noted.

File: Back-End/controllers/instructors_controllers/courses_controllers.js (hunk 3 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/instructors_controllers/courses_controllers.js]: Noted.

File: Back-End/controllers/instructors_controllers/courses_controllers.js (hunk 4 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/instructors_controllers/courses_controllers.js]: Noted.

File: Back-End/routes/admin.route.js (hunk 5 above)
Comment (Omar-Alkhatib) [reply Back-End/routes/admin.route.js]: Noted.

File: Back-End/controllers/admin_controllers/instructors_controllers.js (hunk 6 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/admin_controllers/instructors_controllers.js]: Point taken, but I'm referring to a specific id in a table. How can I make that dynamic?

File: Back-End/controllers/admin_controllers/students_controllers.js (hunk 7 above)
Comment (Omar-Alkhatib) [reply Back-End/controllers/admin_controllers/students_controllers.js]: Point taken, but I'm referring to a specific id in a table. How can I make that dynamic?

**Pull request merged**